saved-sites/saved-sites-impl/src/main/java/com/duckduckgo/savedsites/impl/sync/algorithm/SavedSitesTimestampPersister.kt

```diff
@@ -50,7 +51,7 @@ class SavedSitesTimestampPersister @Inject constructor(
                     Timber.d("Sync-Bookmarks-Persister: folder ${folder.id} modified before local folder, nothing to do")
                 } else {
                     Timber.d("Sync-Bookmarks-Persister: folder ${folder.id} modified after local folder, replacing content")
-                    savedSitesRepository.replaceFolderContent(folder, localFolder.id)
+                    syncSavedSitesRepository.replaceFolder(folder, children)
```

Comment:
something that confuses me across the strategies is why when remote wins, if the entity exists, we do replace relations with new children list. But if it doesn't exist, we only insert the folder without children. 

I know, when the folder doesn't exist, we know the children relations will be created later when processing each children entity in the payload. Having 2 different behaviors in the same method, with the assumption of knowing what happens outside this method, it's telling me something is wrong here.

Reply:
it can definitely be improved, but I don’t think that we need to take care of this now. one of the first things we need to do is improve this algorithm as per https://app.asana.com/0/0/1206000285476376/f